numpy/ma/extras.py

```diff
@@ -716,6 +716,54 @@ def median(a, axis=None, out=None, overwrite_input=False):
 
 
 #..............................................................................
+def compress_nd(x, axis=None):
+    """Supress slices from multiple dimensions which contain masked values.
+    
+    Parameters
+    ----------
+    x : array_like, MaskedArray
+        The array to operate on. If not a MaskedArray instance (or if no array
+        elements are masked, `x` is interpretted as a MaskedArray with `mask`
+        set to `nomask`.
+    axis : tuple of ints or int, optional
+        Which dimensions to supress slices from can be configured with this
+        parameter.
+        - If axis is a tuple of ints, those are the axes to supress slices from.
+        - If axis is an int, then that is the only axis to supress slices from.
+        - If axis is None, all axis are selected.
+    
+    Returns
+    -------
+    compress_array : ndarray
+        The compressed array.        
+    """    
+    x = asarray(x)
+    m = getmask(x)
+    # Set axis to tuple of ints
+    if axis is None:
+        axis = range(len(x.shape))
+    elif isinstance(axis, int):
+        axis = (axis % x.ndim,)
+    else:
+        axis = tuple(ax % x.ndim for ax in axis)
+    # Check axis input
+    for ax in axis:
+        if not (0 <= ax < x.ndim):
```

Comment:
This test will never fail, because you have taken the modulo `x.ndim` of all the entries in `axis` right before. Typically, only axis such that `-x.ndim <= ax < x.ndim` are accepted, and negative axes are converted to positive by adding `x.ndim` to them. But an array of e.g. 3 dimensions should not accept an axis of -4 or +3...